TBG.Business/SingleEliminationTournament.cs

```diff
@@ -9,14 +9,124 @@ namespace TBG.Business
-        public int TournamentTypeId { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public List<ITeam> Participants { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public List<ITournamentPrize> Prizes { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        List<ITournamentEntry> ITournament.Participants { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public int TournamentId { get; set; }
+        public int UserId { get; set; }
+        public string TournamentName { get; set; }
+        public decimal EntryFee { get; set; }
+        public double TotalPrizePool { get; set; }
+        public int TournamentTypeId { get; set; }
+        public List<ITournamentEntry> Participants { get; set; }
+        public List<ITournamentPrize> Prizes { get; set; }
+        public List<IRound> Rounds { get; set; }
+
+        public SingleEliminationTournament()
+        {
+            Participants = new List<ITournamentEntry>();
+            Prizes = new List<ITournamentPrize>();
+            Rounds = new List<IRound>();
+        }
+
+        public bool BuildTournament()
+        {
+            Queue<ITournamentEntry> teamQueue = new Queue<ITournamentEntry>();
+            foreach (var team in Participants)
+            {
+                teamQueue.Enqueue(team);
+            }
+
+            for (int i = 1; i <= CalculateRoundTotal(Participants.Count); i++)
+            {
+                Rounds.Add(new Round()
+                {
+                    RoundNum = i,
+                    TournamentId = TournamentId
+                });
+            }
+
+            foreach (var round in Rounds)
+            {
+                Matchup nextRoundMatchup = new Matchup();
+
+                if (round.RoundNum == 1)
+                {
+                    for (int i = 0; i < Participants.Count / 2; i++)
+                    {
+                        round.Pairings.Add(new Matchup()
+                        {
+                            MatchupId = i,
+                            Teams = new List<IMatchupEntry>()
+                        });
+                    }
+
+                    foreach (var matchup in round.Pairings)
+                    {
+                        matchup.Teams.Add(new MatchupEntry()
+                        {
+                            TheTeam = teamQueue.Dequeue(),
+                            Score = 0
+                        });
+
+                        matchup.Teams.Add(new MatchupEntry()
+                        {
+                            TheTeam = teamQueue.Dequeue(),
+                            Score = 0
+                        });
+                    }
+                }
+
+                if (round.Pairings.Count > 1)
```

Comment:
Just basically this If statement creates matchups in the next rounds if there are enough teams left?

Reply:
Yes, it creates matchups in the next round if there are two or more pairings in the current round to feed the next round.